api/enterprise-api-ref/personal-groups.markdown

```diff
@@ -0,0 +1,395 @@
+  }
+}
+```
+
+
+**Operators:**
+
+For filtering you can use the operators below:
+
+| Operator        |
+|-----------------|
+| <               |
+| >               |
+| =               |
+| !=              |
+| <=              |
+| >=              |
+| matches         |
+| not_match       |
+| contains        |
+| not_contain     |
+| regex_matches   |
+| regex_not_match |
+| is_reported     |
+| is_not_reported |
+
+
+```
+curl -k --user <username>:<password> \
+  -X POST \
+  https://hub.cfengine.com/api/host-groups/personal/ \
+  -H 'content-type: application/json' \
+  -d '{
+        "name":"AIX hosts with additional SHA=first1 host",
+        "description":"Host name",
+        "filter":{
+            "filter": {
+                "Architecture": {
+                    "matches": "86"
+                }
+            },
+            "hostsFilter": {
+                "includes": {
+                    "type": "hostkey",
+                    "data": [
+                        "SHA=first1"
+                    ]
+                }
+            },
+            "hostContextInclude": [
+                "aix"
+            ]
+        }
+  }'
+```
+
+
+**Example response:**
+
+```json
+{"id":"4"}
+```
+
+## Update group
+
+**URI:** https://hub.cfengine.com/api/host-groups/personal/:id
+
+**Method:** PATCH
+
+**Parameters:**
+
+* **id** *(integer)*
+  Group id.
+* **name** *(string)*
+  Group name.
+
+* **description** *(string)*
+  Group description.
+
+* **filter** *(json object)*  Group filter object. Includes inventory filter and classes filters
+    * **filter** *(json object)*  Optional parameter.
+      Inventory filter data. You can use array values for multiple filter, the logic will be AND. Format is
+    * **hostContextInclude** *(array)* Optional parameter.
+      Includes only results that concern hosts which have all specified CFEngine contexts (class) set.
+    * **hostContextExclude** *(array)*  Optional parameter.
+      Excludes results that concern hosts which have specified CFEngine context (class) set. Hosts that have at least one
+      of the specified contexts set will be excluded from the results.
+  * **hostsFilter** *(json object)* Optional parameter.
+      * **includes** *(json object)* Optional parameter.
+        Object that specifies hosts to be included.
+          * **includeAdditionally** *(boolean)* Default: `false`
+            Defines if hosts will be included additionally on top of the inventory filters or class filters.
+          * **type** (string)
+            Now the only supported type is `hostkey`.
+          * **data** *(array)*
+            An array of hosts keys.
+      * **excludes** *(json object)* Optional parameter.
+        Object that specifies hosts to be excluded.
+          * **type** (string)
+            Now the only supported type is `hostkey`.
+          * **data** *(array)*
+            An array of hosts keys.
+
+```json
+{ 
+  "filter": {
+      "filter":{
```

Comment:
This filter.filter thing was a bit confusing to me. For whatever reason I wasn't seeing both, making me think that hostsFilter and hostcontextinclude/exclude were outside of filter.